Tools/Scripts/webkitpy/common/net/bugzilla/results_fetcher.py

```diff
@@ -207,3 +208,28 @@ def lookup_ews_results_from_bugzilla(
         raise RuntimeError(msg)
 
     return lookup_ews_results_from_bugzilla_patch(patch, bot_filter_name)
+
+
+def lookup_ews_results_from_pr(pr):
+    urls = [status.url for status in pr.statuses]
+    return lookup_ews_results(urls)
+
+
+def lookup_ews_results_from_repo(repo_path):
```

Comment:
The mocks just assume we have `repository` as an argument, right? So maybe we should do that, and construct a `repository` object from the CLI `main`?